types/snarkjs/index.d.cts

```diff
@@ -0,0 +1,244 @@
+export as namespace snarkjs;
+
+// Some types borrowed from https://github.com/erhant/circomkit.
+
+// A signal value is a number, or an array of numbers (recursively).
+export type SignalValueType = `${number}` | number | bigint | SignalValueType[];
+
+// An object with string keys and array of numerical values.
+// Each key represents a signal name as it appears in the circuit.
+export interface CircuitSignals {
+    [signal: string]: SignalValueType;
+}
+
+// Some fields for the R1CS information returned by SnarkJS.
+// Many other fields are omitted in this type.
+export interface R1CSInfoType {
+    variables: number;
+    constraints: number;
+    privateInputs: number;
+    publicInputs: number;
+    labels: number;
+    outputs: number;
+    prime: bigint;
+    primeName: string;
+}
+
+export interface Groth16Proof {
+    pi_a: string[];
+    pi_b: string[][];
+    pi_c: string[];
+    protocol: string;
+    curve: string;
+}
+
+export interface FflonkProof {
+    polynomials: {
+        C1: string[];
+        C2: string[];
+        W1: string[];
+        W2: string[];
+    };
+    evaluations: object;
+    protocol: string;
+    curve: string;
+}
+
+export interface PlonkProof {
+    A: string[];
+    B: string[];
+    C: string[];
+    Z: string[];
+    T1: string[];
+    T2: string[];
+    T3: string[];
+    Wxi: string[];
+    Wxiw: string[];
+    eval_a: string;
+    eval_b: string;
+    eval_c: string;
+    eval_s1: string;
+    eval_s2: string;
+    eval_zw: string;
+    protocol: string;
+    curve: string;
+}
+
+export type PublicSignals = string[];
+
+export namespace groth16 {
+    function exportSolidityCallData(_proof: Groth16Proof, _pub: PublicSignals): Promise<string>;
+    function fullProve(
+        _input: CircuitSignals,
+        wasmFile: string,
+        zkeyFileName: string,
+        logger?: any,
+    ): Promise<{
+        proof: Groth16Proof;
+        publicSignals: PublicSignals;
+    }>;
+    function prove(
+        zkeyFileName: string,
+        witnessFileName: any,
```

Comment:
It looks like it can be constrained with `string` type.

Reply:
I'm not really sure about it, because the second parameter here -> https://github.com/iden3/snarkjs/blob/33c753b1c513747e3b1f2d6cab2ca2e0c830eb77/src/groth16_fullprove.js#L32 is not a string. Maybe it can _also_ be a string.

Reply:
You're right, it's a bit messy imo.

Reply:
> You're right, it's a bit messy imo.

Yea, let's leave it as it is. If necessary, it can be updated.

Reply:
Exactly!